{{ cookiecutter.repo_name }}/plugins/{{ cookiecutter.plugin_name }}/{%- if cookiecutter.ui_type != "none" %}UI{{ cookiecutter.plugin_name }}.cpp{%- endif %}

```diff
@@ -0,0 +1,99 @@
+/*
+ * {{ cookiecutter.project_name }} audio effect based on DISTRHO Plugin Framework (DPF)
+ * Copyright (C) {{ cookiecutter.year }} {{ cookiecutter.author_name }} <{{ cookiecutter.email }}>
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+ * SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+ * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+ * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include "UI{{ cookiecutter.plugin_name }}.hpp"
+
+START_NAMESPACE_DISTRHO
+
+UI{{ cookiecutter.plugin_name }}::UI{{ cookiecutter.plugin_name }}()
+: UI(600, 400)  {
+
+}
+
+UI{{ cookiecutter.plugin_name }}::~UI{{ cookiecutter.plugin_name }}() {
+
+}
+
+void UI{{ cookiecutter.plugin_name }}::parameterChanged(uint32_t index, float value) {
+    fParams[index] = value;
+
+    switch (index) {
+        case Plugin{{ cookiecutter.plugin_name }}::paramVolume:
+            // do something when volume param is set
+            break;
+    }
+}
+
+void UI{{ cookiecutter.plugin_name }}::programLoaded(uint32_t index) {
+    switch (index) {
+        case 0:
+            fParams[PluginSimpleGain::paramVolume] = 0.1f;
```

Comment:
This seems a bit pointless on it's own. Maybe instead insert a comment like this:

    // Set values for each parameter and update their widgets

I'm not sure, would it be ok, to call `parameterChanged` here?

Reply:
Yes, I think calling parameterChanged makes sense here.